sql/handler.cc

```diff
@@ -8566,6 +8566,7 @@ bool handler::is_using_prohibited_gap_locks(TABLE* table,
   if (!using_full_primary_key
       && has_transactions()
       && !has_gap_locks()
+      && thd_tx_isolation(thd) >= ISO_REPEATABLE_READ
```

Comment:
Yeah, something is wrong here, unsure of what, likely a GCA and working branch mixup when I forward merged this to 5.7.